crates/red_knot_python_semantic/src/types.rs

```diff
@@ -397,103 +261,22 @@ impl ScopedTypeId for ScopedIntersectionTypeId {
 // case where a Not appears outside an intersection (unclear when that could even happen, but we'd
 // have to represent it as a single-element intersection if it did) in exchange for better
 // efficiency in the within-intersection case.
-#[derive(Debug, PartialEq, Eq, Clone)]
+#[salsa::interned]
 pub struct IntersectionType<'db> {
     // the intersection type includes only values in all of these types
-    positive: FxIndexSet<Type<'db>>,
+    positive: Vec<Type<'db>>,
     // the intersection type does not include any value in any of these types
-    negative: FxIndexSet<Type<'db>>,
-}
-
-#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
-pub struct ScopedModuleTypeId;
-
-impl ScopedTypeId for ScopedModuleTypeId {
-    type Ty<'db> = ModuleType;
-
-    fn lookup_scoped<'a, 'db>(self, types: &'a TypeInference<'db>) -> &'a Self::Ty<'db> {
-        types.module_ty()
-    }
+    negative: Vec<Type<'db>>,
 }
 
-impl<'db> TypeId<'db, ScopedModuleTypeId> {
-    fn member(self, context: &TypingContext<'db, '_>, name: &Name) -> Option<Type<'db>> {
-        context.public_symbol_ty(self.scope.file(context.db), name)
-    }
-}
-
-#[derive(Debug, Eq, PartialEq, Clone)]
-pub struct ModuleType {
+#[salsa::interned]
+pub struct ModuleType<'db> {
     file: VfsFile,
 }
```

Comment:
I'm unsure if it is worth interning `ModuleType`, considering that it only stores a `VfsFile`. A salsa ingredient requires more metadata than the amount of memory that we save by interning the module type.

Reply:
Good point, I can try removing this.

Reply:
I just removed `ModuleType` entirely in favor of having `Type::Module` directly hold a `VfsFile`. At the moment there's really no reason to have the extra `ModuleType` layer at all, just to implement the `member` method, which is a one-liner that can just as well go directly in `Type::member` in the `Type::Module` arm.